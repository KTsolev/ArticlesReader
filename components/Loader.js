import React from "react";
import { View, ActivityIndicator } from "react-native";
import { headerConfiguration } from './Header';
import { WebView } from 'react-native-webview';

class Loader extends React.Component {
    static navigationOptions = headerConfiguration;
    render() {
      return (
        <View style={{
            justifyContent: 'center',
            alingItems: 'center',
            position: 'absolute',
            zIndex:1,
            top: 0, 
            bottom: 0, 
            left: 0, 
            right: 0, 
            backgroundColor:'rgba(0, 179, 149, 0.35)'}}>
            <ActivityIndicator size="large" color="#fff" />
        </View>
      );
    }
  }
 
export default Loader;