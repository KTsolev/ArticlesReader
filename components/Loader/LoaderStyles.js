import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: { justifyContent: 'center', alignItems: 'center', position: 'absolute', 
        zIndex:1, top: 0, bottom: 0, left: 0, right: 0, 
        backgroundColor:'rgba(0, 179, 149, 0.35)' }
});