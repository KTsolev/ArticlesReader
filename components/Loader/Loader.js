import React from 'react';
import { View, ActivityIndicator } from 'react-native';
import { headerConfiguration } from '../Header/Header';
import { styles } from './LoaderStyles';

class Loader extends React.Component {
    static navigationOptions = headerConfiguration;
    render() {
      const { container } = styles;
      return (
        <View style={container}>
            <ActivityIndicator size="large" color="#fff" />
        </View>
      );
    }
  }
 
export default Loader;