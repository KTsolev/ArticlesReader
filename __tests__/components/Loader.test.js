import Loader from '../../components/Loader/Loader';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<Loader />', () => {
    test('renders the component', () => {
        const wrapper = shallow(<Loader />);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });
});