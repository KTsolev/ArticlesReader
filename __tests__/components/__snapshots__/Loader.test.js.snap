// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders the component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 179, 149, 0.35)",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#fff"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
